webpack.config.js

```diff
@@ -13,34 +28,59 @@ const babelRule = {
           'env',
           {
             targets: {
-              browsers: ['last 2 versions']
+              browsers: ['last 2 versions'],
             },
-            modules: false
-          }
-        ]
-      ]
-    }
-  }
+            modules: false,
+          },
+        ],
+      ],
+    },
+  },
 };
 
+const outputPath = path.resolve(__dirname, 'public/js');
+
 const webpackConfig = {
   devtool: 'source-map',
+  context: path.resolve(__dirname, 'src/js'),
+  entry: entries,
+  resolve: {
+    extensions: ['.js', '.json', '.jsx'],
+  },
+  output: {
+    filename: '[name].js',
+    path: outputPath,
+    publicPath: '/js/',
+  },
   module: {
     rules: [
       babelRule,
       {
         test: /\.json$/,
-        loader: 'json-loader'
-      }
-    ]
+        loader: 'json-loader',
+      },
+    ],
   },
   plugins: [
+    new CleanWebpackPlugin([outputPath]),
+    new webpack.ProvidePlugin({
+      $: 'jquery',
+      jQuery: 'jquery',
+      jquery: 'jquery',
+      d3: 'd3',
+    }),
     // To extract node_modules imports into a vendor.js file.
     new webpack.optimize.CommonsChunkPlugin({
       name: 'vendor',
-      minChunks: module => module.context && module.context.includes('node_modules')
-    })
-  ]
+      minChunks: module => module.context && module.context.includes('node_modules'),
+    }),
+  ],
```

Comment:
just using a more vanilla version of airbnb. Lots of file errors still so it's mostly for simple guidance.